pkg/apis/openebs.io/scp/v1alpha1/storage-cohort.go

```diff
@@ -27,10 +27,277 @@ import (
 }
 
+// StorageCohortDataProtectionCapabilities
+type StorageCohortDataProtectionCapabilities struct {
+	// Whether replication is supported by the cohort
+	// +optional
+	Replication *bool `json:"replication,omitempty"`
+
+	// Backup controls data can be backed up by the cohort
+	// +optional
+	Backup *bool `json:"backup,omitempty"`
+
+	// whether cohort will support snapshots
+	// +optional
+	Snapshots *bool `json:"snapshots,omitempty"`
+}
+
+// ComponentCondition contains condition information for a storage cohort.
+type ComponentCondition struct {
+	// Name of the component
+	// For example: "cohort-manager"
+	Name string `json:"name"`
+	// Type of component condition.
+	Type ComponentConditionType `json:"type"`
+	// Status of the condition, one of True, False, Unknown.
+	Status ComponentConditionStatus `json:"status"`
+	// Last time we got an update on a given condition.
+	LastHeartbeatTime metav1.Time `json:"lastHeartbeatTime,omitempty"`
+	// Last time the condition transit from one status to another.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+	// (brief) reason for the condition's last transition.
+	Reason string `json:"reason,omitempty"`
+	// Human readable message indicating details about last transition.
+	Message string `json:"message,omitempty"`
+}
+
+type StorageCohortEncryptionAlgorithm string
+
+// These are valid encryption algorithms.
+const (
+	// TODO: verify the algorithms required
+	AES StorageCohortEncryptionAlgorithm = "AES"
+	DES StorageCohortEncryptionAlgorithm = "DES"
+	ECC StorageCohortEncryptionAlgorithm = "ECC"
+	RSA StorageCohortEncryptionAlgorithm = "RSA"
+)
+
+type StorageCohortDataSanitizationPolicy string
+
+const (
+	NoneDataSanitizationPolicy               StorageCohortDataSanitizationPolicy = "None"
+	ClearDataSanitizationPolicy              StorageCohortDataSanitizationPolicy = "Clear"
+	CryptographicEraseDataSanitizationPolicy StorageCohortDataSanitizationPolicy = "CryptographicErase"
+)
+
+type ComponentConditionType string
+
+// These are valid conditions of cohort.
+// In the future, we can add more. The current set of conditions are:
+// CohortComponentReady, CohortComponentNotReady.
+const (
+	// CohortComponentReady means cohort component is healthy and ready to perform its task.
+	CohortComponentReady ComponentConditionType = "Ready"
+	// CohortComponentNotReady means the cohort component is unhealthy and not able to perform its task.
+	CohortComponentNotReady ComponentConditionType = "NotReady"
+	// TODO add more types if necessary
+)
+
+type ComponentConditionStatus string
+
+// These are valid condition statuses. "ConditionTrue" means a component is in the condition.
+// "ConditionFalse" means a component is not in the condition. "ConditionUnknown" means
+// cohort-operator/cohort-manager can't decide if a cohort component is in the condition or not.
+const (
+	ConditionTrue    ComponentConditionStatus = "True"
```

Comment:
What is "in the condition"?

Reply:
One example would be:
```
{
          {
                "lastHeartbeatTime": "2021-09-01T10:34:59Z",
                "lastTransitionTime": "2021-06-28T16:15:00Z",
                "message": "Cohort manager is posting ready status.",
                "reason": "CohortComponentReady",
                "status": "True",
                "type": "Ready"
            },
            {
                "lastHeartbeatTime": "2021-09-01T10:34:59Z",
                "lastTransitionTime": "2021-06-28T16:15:00Z",
                "message": "Cohort manager is unreachable.",
                "reason": "CohortComponentNotReady",
                "status": "True",
                "type": "NotReady"
            }
```
So, the status `True` here  means that the cohort is in this condition. The conditions is basically a array and can have predefined conditions filled inside it. And through this `status` field we'll get to know whether the cohort is in that condition or not.

Reply:
Yes, but for better information about the cohort and debugging these can be used. This is the way a node's status is shown by k8s also.